safehttp/header.go

```diff
@@ -0,0 +1,134 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package safehttp
+
+import (
+	"net/http"
+	"net/textproto"
+)
+
+var disallowedHeaders = map[string]bool{"Set-Cookie": true}
+
+// Header represents the key-value pairs in an HTTP header.
+// The keys will be in canonical form, as returned by
+// textproto.CanonicalMIMEHeaderKey.
+type Header struct {
+	wrappedHeader http.Header
```

Comment:
Maybe this can be an anonymous struct member. This means that alll methods of http.Header would be available in this Header type, unless you overwrite them.

Reply:
At the moment there are three methods in `http.Header` (`Write`, `WriteSubset` and `Clone`) that we don't know if we want to have in `safehttp.Header`. We have decided to not have them right now. So unless there is a way of excluding some of the methods from `http.Header` while having it as an anonymous struct field then we can't do this right now.

Reply:
Makes sense. Thanks.